utils.h

```diff
@@ -100,22 +107,17 @@ bool is_special_ver_ndx(Elf64_Versym v);
 std::string special_ver_ndx_to_str(Elf_Versym v);
 
 template <class T>
-inline std::string HexString(T num, int length = 16) {
+inline std::string HexString(T num, int length = -1) {
+    if (length == -1) {
+        length = sizeof(T) * 2;
+    }
     std::stringstream ss;
-    ss << "0x" << std::uppercase << std::setfill('0') << std::setw(length) << std::hex << num;
+    ss << "0x" << std::uppercase << std::setfill('0') << std::setw(length) << std::hex << +num;
```

Comment:
Why `+` is added?

Reply:
This is to print `char` or `*char` data as a number.

Reply:
Ah, it's a nice trick :)